ibis/backends/pandas/client.py

```diff
@@ -139,9 +142,16 @@ def _infer_pandas_series_contents(s: pd.Series) -> dt.DataType:
         if inferred_dtype == 'mixed':
             # We need to inspect an element to determine the Ibis dtype
             value = s.iloc[0]
-            if isinstance(value, (np.ndarray, list, pd.Series)):
+            if isinstance(value, (np.ndarray, pd.Series, Sequence)):
                 # Defer to individual `infer` functions for these
                 return dt.infer(value)
+            elif isinstance(value, Mapping):
```

Comment:
Shall we implement this in `expr/datatypes` instead?

Reply:
Done.